### ISSUE TITLE
Some folks not seeing sensors/switches in homeassistant

### ISSUE THREAD

**Issue by krk628** [krk628/hassio-screenlogic-addon#1]

**Comment by krk628** [comment]
@pgregg88 said...

1. I followed your instructions and successfully connected to my SL controller from Raspi with node.js
2. I added the referenced switches and sensors into the appropriate HA .yaml files on my hassio instance.
3. I successfully cloned and installed your Pentair Local Add-on within HA.
4. I can view the new mqtt queues (/pentair/pooltemp/state, etc.) and they are receiving frequent messages. Sweet!

However, I'm failing to pick up the sensors within HA.

-- snip -- snip -- snip --

**Comment by krk628** [comment]
yeah the Screenlogic->MQTT_Broker part is working so more than 1/2 way there.   the MQTT sensors/switches should be the easy part.  make sure HA gets restarted for sure; and you should see them under http://hassio.local:8123/dev-state (or however you access your hass.io)

**Comment by pgregg88** [comment]
Success! I needed to enable mosquitto mqtt as an HA add-on (instead of my remote mqtt broker) and set "anonymous": true, (default is false) in the ad-on config

![image](https://user-images.githubusercontent.com/4943027/53224906-e71e3e00-363b-11e9-98db-361e831547de.png)

![image](https://user-images.githubusercontent.com/4943027/53225067-690e6700-363c-11e9-8431-11f0db6205e5.png)

**Comment by krk628** [comment]
cool; so works for both of us.  again, i just whipped this up in an hour so it can be improved.  for example I think the addon will hang if the screenlogic interface on the pool is down.  but anyways, as to the MQTT issues it all happens in run.sh.   I'm just assuming an anonymous broker connection but you can change line 14:
PAYLOAD=`mosquitto_sub -v -t /pentair/# -W 10 -C 1`
to
PAYLOAD=`mosquitto_sub -u myusername -p mypassword -v -t /pentair/# -W 10 -C 1`

and change line 43:
node send_state_to_ha.js | awk -F, '{print "mosquitto_pub -t " $1 " -m " $2}' | bash -s
to:
node send_state_to_ha.js | awk -F, '{print "mosquitto_pub -u myusername -p mypassword -t " $1 " -m " $2}' | bash -s


i run ubuntu-mate not hassio; but it looks like you can feed a username, password, and other MQTT broker parameters to run.sh with addon config variables...just haven't looked at how to do it yet.   but in theory the addon could operate as A) an anonymous MQTT connection to localhost or B) a remote MQTT broker w/optional username/password.     traveling again but i'll look in a few days.

**Comment by pgregg88** [comment]
Cool. I’ll experiment with connecting to an external MQTT broker. I also want to get control of additional circuits. Thank you for getting this rolling.

**Comment by krk628** [comment]
if you know hassio please go for it and send me pull.   and yeah...i have two pumps on circuits 6 and 7 for acid and chlorine so it'll be necessary to be able to control those too.   and if anybody knows how to control lights do tell :)

**Comment by parnic** [comment]
You can control the lights with setCircuitState, same as the pool and spa. The circuit IDs come back from getPoolStatus() in the circuitArray.

**Comment by pgregg88** [comment]
@parnic Sweet! that's what I was hoping for. Thx.

**Comment by bertfaerstain** [comment]
so no programmer type here...  I installed mqtt, installed the pentair
module and it shows it's yellow (running), but I don't even see how to
enter the IP address of my pentair controller.  I think I don't have
everything installed/working correctly.   Sugs appreciated..

**Comment by pgregg88** [comment]
@bertfaerstain The first version assumes you’re running the hassio embedded MQTT broker with anonymous = true (meaning no username or password required). 

The next version has the ability to enter an IP address, username and password for the MQTT broker. Hang tight. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by krk628** [comment]
@bertfaerstain  Yeah hang tight.  This first go was just to assess feasibility so only broadcast connections were supported.  I'm adding direct and remote connections (remote is new as of node-screenlogic-1.2 a couple days ago), a more complete MQTT integration (authentication, ssl, and non-localhost brokers), and adding as much of the node-screenlogic bits as I can figure out.   should be a day or two.  k

**Comment by bertfaerstain** [comment]
thanks for the update guys... appreciate it.   one more thing to add :
(pure conjecture btw) but is there a way to design/code this without mqtt?
SmartThings access via mqtt was often really wonky when I had that path
connected, but now that hassio has native smarthings api support (in ver
88.x) it's now WAAYY more reliable and interactive.  So I was wondering if
mqtt just complicates things when there's a viable way to connect hassio to
pentair without mqtt.  (just a thought)

**Comment by krk628** [comment]
@bertfaerstain Concur, MQTT was just the most obvious way for me to do it at the time as I don't even run HASSIO honestly...I use command_line switches/sensors.  It's one of those GoodthingsBadthings with homeassistant that there's so many ways to do things...and it gets messy with HASSIO as you have all the Docker-related constraints.     Do I understand that you are running HASSIO?
